internal/pkg/template/templates/environment/partials/cdn-resources.yml

```diff
@@ -17,8 +17,40 @@ CloudFrontDistribution:
             OriginProtocolPolicy: match-viewer # Will need to change with TLS termination and Alias usage
           DomainName: !GetAtt PublicLoadBalancer.DNSName
           Id: !Sub 'copilot-${AppName}-${EnvironmentName}-origin'
-      # ViewerCertificate: # Used for Aliases
-      #   AcmCertificateArn: # Imported certificate goes here
-      #   CloudFrontDefaultCertificate: true # If importing certificate, set to false
-      #   MinimumProtocolVersion: TLSv1 # Default for now, may need to change later
-      #   SsLSupportMethod: sni-only # Changing this can incur costs to the user
+{{- if not .PublicHTTPConfig.ImportedCertARNs}}
+      ViewerCertificate:
+        !If
+          - DelegateDNS
+          - AcmCertificateArn: !Ref CertificateReplicator
+            MinimumProtocolVersion: TLSv1
+            SslSupportMethod:  sni-only
+          - !Ref AWS::NoValue
+
+CertificateReplicatorFunction:
+  Type: AWS::Lambda::Function
+  Condition: DelegateDNS
+  Properties:
+    {{- with $cr := index .CustomResources "CertificateReplicatorFunction" }}
+    Code:
+      S3Bucket: {{$cr.Bucket}}
+      S3Key: {{$cr.Key}}
+    {{- end}}
+    Handler: "index.certificateReplicateHandler"
+    Timeout: 900
+    MemorySize: 512
+    Role: !GetAtt 'CustomResourceRole.Arn'
+    Runtime: nodejs12.x
+    
+CertificateReplicator:
+  Metadata:
+    'aws:copilot:description': 'Replicate certificate to us-east-1'
```

Comment:
Can we create these resources on the condition that the `AWS::Region` isn't us-east-1?

Reply:
I think it makes more sense to have a resource which represents "the certificate used to validate cloudfront" rather than potentially having a certificate which is created in the specific case it's needed. The drawback here is that we may create more resources than we strictly need, but I think the simplifications to the code are worth it, especially that it helps allow the user to understand what we're using the stack for better.